monkey/agent_plugins/exploiters/log4shell/src/plugin.py

```diff
@@ -0,0 +1,116 @@
+import logging
+from functools import partial
+from pprint import pformat
+from typing import Any, Dict, List, Sequence, Set
+
+# common imports
+from common.event_queue import IAgentEventPublisher
+from common.types import AgentID, Event, NetworkPort
+from common.utils.code_utils import del_key
+
+# dependencies to get rid of or internalize
+from infection_monkey.exploit import IAgentBinaryRepository, IAgentOTPProvider
+from infection_monkey.exploit.tools import filter_out_closed_ports, get_open_http_ports
+from infection_monkey.exploit.tools.http_agent_binary_server import start_agent_binary_server
+from infection_monkey.i_puppet import ExploiterResultData, TargetHost
+from infection_monkey.network import TCPPortSelector
+
+# from .log4shell_exploit_client import Log4ShellExploitClient
+# from .log4shell_exploiter import Log4ShellExploiter
+from .log4shell_options import Log4ShellOptions
+
+logger = logging.getLogger(__name__)
+
+
+class Placeholder_Log4ShellExploitClient:
+    def __init__(self, *args, **kwargs):
+        pass
+
+
+class Placeholder_Log4ShellExploiter:
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def exploit_host(self, *args, **kwargs):
+        pass
+
+
+def get_ports_to_exploit(
+    host: TargetHost, target_ports: List[NetworkPort], try_all_discovered_http_ports: bool
+) -> Set[NetworkPort]:
+    ports_to_exploit = filter_out_closed_ports(host, target_ports)
+    if try_all_discovered_http_ports:
+        ports_to_exploit.update(get_open_http_ports(host))
+
+    return ports_to_exploit
+
+
+def should_attempt_exploit(ports_to_exploit: Set[NetworkPort]) -> bool:
+    return bool(ports_to_exploit)
```

Comment:
```suggestion
def should_attempt_exploit(
    host: TargetHost, target_ports: List[NetworkPort], try_all_discovered_http_ports: bool
) -> Set[NetworkPort]:
    ports_to_exploit = filter_out_closed_ports(host, target_ports)
    if try_all_discovered_http_ports:
        ports_to_exploit.update(get_open_http_ports(host))

    return bool(ports_to_exploit)

```

Reply:
`get_ports_to_exploit()` is being used later in the file too.